Templates/AutoMockable.stencil

```diff
@@ -9,17 +11,23 @@ class {{ type.name }}Mock: {{ type.name }} {
     {% for method in type.allMethods %}
     //MARK: - {{ method.shortName }}
 
-    {% if not method.shortName == "init" %}var {{ method.shortName }}Called = false{% endif %}
+    {% if not method.isInitializer %}var {{ method.shortName }}Called = false{% endif %}
     {% if method.parameters.count == 1 %}var {{ method.shortName }}Received{% for param in method.parameters %}{{ param.name|upperFirst }}: {{ param.typeName.unwrappedTypeName }}?{% endfor %}{% else %}{% if not method.parameters.count == 0 %}var {{ method.shortName }}ReceivedArguments: ({% for param in method.parameters %}{{ param.name }}: {% if param.typeAttributes.escaping %}{{ param.unwrappedTypeName }}{% else %}{{ param.typeName }}{% endif %}{% if not forloop.last %}, {% endif %}{% endfor %})?{% endif %}{% endif %}
-    {% if not method.returnTypeName.isVoid and not method.shortName == "init" %}var {{ method.shortName }}ReturnValue: {{ method.returnTypeName }}!{% endif %}
+    {% if not method.returnTypeName.isVoid and not method.isInitializer %}var {{ method.shortName }}ReturnValue: {{ method.returnTypeName }}!{% endif %}
 
-    func {{ method.shortName }}({% for param in method.parameters %}{% if param.argumentLabel == nil %}_{% else %}{{ param.argumentLabel }}{% endif %}{% if not param.argumentLabel == param.name %} {{ param.name }}{% endif %}: {{ param.typeName }}{% if not forloop.last %}, {% endif %}{% endfor %}){% if not method.returnTypeName.isVoid %} -> {{ method.returnTypeName }}{% endif %} {
+    {% if method.isInitializer %}
+    required {{ method.name }} {
 
-        {% if not method.shortName == "init" %}{{ method.shortName }}Called = true{% endif %}
-        {%if method.parameters.count == 1 %}{{ method.shortName }}Received{% for param in method.parameters %}{{ param.name|upperFirst }} = {{ param.name }}{% endfor %}{% else %}{% if not method.parameters.count == 0 %}{{ method.shortName }}ReceivedArguments = ({% for param in method.parameters %}{{ param.name }}: {{ param.name }}{% if not forloop.last%}, {% endif %}{% endfor %}){% endif %}{% if not method.returnTypeName.isVoid %}{% endif %}
-        {% if not method.returnTypeName.isVoid and not method.shortName == "init" %}return {{ method.shortName }}ReturnValue{% endif %}{% endif %}
+        {%if method.parameters.count == 1 %}{{ method.shortName }}Received{% for param in method.parameters %}{{ param.name|upperFirst }} = {{ param.name }}{% endfor %}{% else %}{% if not method.parameters.count == 0 %}{{ method.shortName }}ReceivedArguments = ({% for param in method.parameters %}{{ param.name }}: {{ param.name }}{% if not forloop.last%}, {% endif %}{% endfor %}){% endif %}{% endif %}
```

Comment:
can we extract that and the code above for not-initialiser case into macro node (from SwiftStencilKit)? Look very similar to me.